app/forms/hyrax/forms/permission_template_form.rb

```diff
@@ -38,14 +38,61 @@ def initialize(model)
         select_release_varies_option(model)
       end
 
+      # @return [Hash] { :content_tab (for confirmation message) , :updated (true/false), :error_code (for flash error lookup) }
       def update(attributes)
-        update_admin_set(attributes)
-        update_permission_template(attributes)
-        grant_workflow_roles(attributes)
+        return_info = { content_tab: tab_to_update(attributes) }
+        update_status = {}
+        case return_info[:content_tab]
+        when "participants"
+          update_status = update_participants_options(attributes)
+        when "visibility"
+          update_status = update_visibility_options(attributes)
+        when "workflow"
+          update_status = update_workflow_options(attributes)
```

Comment:
If you change the return value from `update_workflow_options`, this block could become 

```ruby
update_workflow_options(attributes)
update_status = { updated: true }
```

Similarly for `update_participants_options`.  

Of the 3 cases, only `update_visibility_options` returns a dynamic response in its Hash, and that is for only two return values.  Even there, the presence of `:error_code` implies `:updated = false`.  So it could just return error_code value or nil.